duckduckgo/macos-browser
# Update GHA release workflow to support enableOnDemand binary

Task/Issue URL: https://app.asana.com/0/1203301625297703/1204915719750897/f

**Description**:
Use enableOnDemand provisioning profile in build_notarized.yml workflow.

**Steps to test this PR**:
1. Verify that [this GitHub Actions run](https://github.com/duckduckgo/macos-browser/actions/runs/5389449997/jobs/9783421087) has completed successfully.
2. Verify that the app uploaded to [this Asana task](https://app.asana.com/0/1203301625297703/1204915719750897/f) works.



---
###### Internal references:
[Pull Request Review Checklist](https://app.asana.com/0/1202500774821704/1203764234894239/f)
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
[Pull Request Documentation](https://app.asana.com/0/1202500774821704/1204012835277482/f)
